src/main/java/customerproductorder/DataContainer.java

```diff
@@ -61,6 +75,17 @@ public DataContainer() {
         return temp;
     }
 
+    public final Customer searchCustomerByCardId(int CardId) {
+        Customer temp = null;
+        for (Customer customer : customers) {
+            if (customer.getCardNumber() == CardId) {
+                temp = customer;
+                break;
+            }
+        }
+        return temp;
```

Comment:
I would not recommend to return NULL. Better use Optional. See for example http://www.oracle.com/technetwork/articles/java/java8-optional-2175753.html

Also it is good to know and use stream operations if you need some filtering logic. See http://www.oracle.com/technetwork/articles/java/ma14-java-se-8-streams-2177646.html